File: app/src/main/java/com/numero/colorpalette/model/ColorStyle.java
package com.numero.colorpalette.model;

import android.support.annotation.StringRes;

import com.numero.colorpalette.R;

public enum ColorStyle {

    RED(
            R.string.red, "RED"
    ),

    PINK(
            R.string.pink, "PINK"
    ),

    PURPLE(
            R.string.purple, "PURPLE"
    ),

    DEEP_PURPLE(
            R.string.deep_purple, "DEEP PURPLE"
    ),

    INDIGO(
            R.string.indigo, "INDIGO"
    ),

    BLUE(
            R.string.blue, "BLUE"
    ),

    LIGHT_BLUE(
            R.string.light_blue, "LIGHT BLUE"
    ),

    CYAN(
            R.string.cyan, "CYAN"
    ),

    TEAL(
            R.string.teal, "TEAL"
    ),

    GREEN(
            R.string.light_green, "LIGHT GREEN"
    ),

    LIME(
            R.string.lime, "LIME"
    ),

    YELLOW(
            R.string.yellow, "YELLOW"
    ),

    AMBER(
            R.string.amber, "AMBER"
    ),

    ORANGE(
            R.string.orange, "ORANGE"
    ),

    DEEP_ORANGE(
            R.string.deep_orange, "DEEP_ORANGE"
    ),

    BROWN(
            R.string.brown, "BROWN"
    ),

    GREY(
            R.string.grey, "GREY"
    ),

    BLUE_GREY(
            R.string.blue_grey, "BLUE GREY"
    );


    private int nameStringRes;
    // FIXME 色モデルに書き換える
    private String color;

    ColorStyle(@StringRes int nameStringRes, String color) {
        this.nameStringRes = nameStringRes;
        this.color = color;
    }

    @StringRes
    public int getNameStringRes() {
        return nameStringRes;
    }

    public String getColor() {
        return color;
    }
}

File: app/src/main/java/com/numero/colorpalette/model/ColorStyle.kt
package com.numero.colorpalette.model

import android.support.annotation.StringRes

import com.numero.colorpalette.R
import com.numero.colorpalette.model.color.*

enum class ColorStyle constructor(@param:StringRes @get:StringRes val nameStringRes: Int, val color: MaterialColor) {

    RED(
            R.string.red, Red()
    ),

    PINK(
            R.string.pink, Pink()
    ),

    PURPLE(
            R.string.purple, Purple()
    ),

    DEEP_PURPLE(
            R.string.deep_purple, DeepPurple()
    ),

    INDIGO(
            R.string.indigo, Indigo()
    ),

    BLUE(
            R.string.blue, Blue()
    ),

    LIGHT_BLUE(
            R.string.light_blue, LightBlue()
    ),

    CYAN(
            R.string.cyan, Cyan()
    ),

    TEAL(
            R.string.teal, Teal()
    ),

    GREEN(
            R.string.green, Green()
    ),

    LIGHT_GREEN(
            R.string.light_green, LightGreen()
    ),

    LIME(
            R.string.lime, Lime()
    ),

    YELLOW(
            R.string.yellow, Yellow()
    ),

    AMBER(
            R.string.amber, Amber()
    ),

    ORANGE(
            R.string.orange, Orange()
    ),

    DEEP_ORANGE(
            R.string.deep_orange, DeepOrange()
    ),

    BROWN(
            R.string.brown, Brown()
    ),

    GREY(
            R.string.grey, Grey()
    ),

    BLUE_GREY(
            R.string.blue_grey, BlueGrey()
    )
}
